cli/src/main/java/com/devonfw/tools/ide/io/FileAccessImpl.java

```diff
@@ -147,24 +144,44 @@ private void downloadFileWithProgressBar(String url, Path target, HttpResponse<I
    */
   private void copyFileWithProgressBar(Path source, Path target) throws IOException {
 
-    try (InputStream in = new FileInputStream(source.toFile());
-        OutputStream out = new FileOutputStream(target.toFile())) {
+    try (InputStream in = new FileInputStream(source.toFile()); OutputStream out = new FileOutputStream(target.toFile())) {
 
       long size = source.toFile().length();
+      informAboutSettingDefaultContentLength(size);
       byte[] buf = new byte[1024];
       int readBytes;
 
       try (IdeProgressBar pb = this.context.prepareProgressBar("Copying", size)) {
         while ((readBytes = in.read(buf)) > 0) {
           out.write(buf, 0, readBytes);
-          pb.stepByOne();
+          if (size > 0) {
+            pb.stepByOne();
+          } else {
+            break;
+          }
         }
+
+        callStepByWithDefaultContentLength(size, pb);
       } catch (Exception e) {
         throw new RuntimeException(e);
       }
     }
   }
 
+  private void callStepByWithDefaultContentLength(long contentLength, IdeProgressBar progressBar) {
+
+    if (contentLength == 0) {
+      progressBar.stepBy(10000000);
+    }
+  }
+
+  private void informAboutSettingDefaultContentLength(long contentLength) {
+
+    if (contentLength == 0) {
+      this.context.warning("Content-Length was not provided by download/copy source. Using fallback: Content-Length for the progress bar is set to 10000000.");
```

Comment:
The default value `10000000` is more or less randomly chosen but there is not big rationale behind it.
If you copy it to multiple places, it would be better to define a constant.

BTW: Before the log statement was also including the URL. IMHO it is helpful to have this information to see where the problem is coming from e.g. if we receive logs via github issues or support.